service/frontend/workflow_handler.go

```diff
@@ -4273,30 +4284,42 @@ func (wh *WorkflowHandler) getHistoryReverse(
 	nextPageToken []byte,
 	branchToken []byte,
 ) (*historypb.History, []byte, int64, error) {
+
+	// readFullPageEventsReverse reads a full page of history events from history service in reverse orcer. Due to
+	// storage format of V2 History it is not guaranteed that pageSize amount of data is returned. Function returns the
+	// list of history events, the size of data read, the next page token, and an error if present.
```

Comment:
There's also several grammatical errors and typos here. Consider rewriting it like this:


readFullPageEventsReverse reads a full page of history events from the history service in reverse order (specify what this means). This function returns the list of history events, the next page token, and an error if present. The history service API will sometimes return fewer results than the requested pageSize, even if there are more results, so we must repeatedly call it until the next page token is nil or we've read a total of pageSize events.

Reply:
same... this is copied from `history_manager_util.go` verbatim for now. I will also reimplement this in a subsequent PR